File: app/src/test/java/mvp/view/game/GamePresenterTest.java
package mvp.view.game;

import android.widget.ViewFlipper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import mvp.model.entity.Card;
import mvp.model.repository.GameRepository;

import static org.mockito.Mockito.verify;

/**
 * Created by Manuel Lorenzo
 */
public class GamePresenterTest {
    @Mock
    private
    Card card;
    @Mock
    private
    ViewFlipper viewFlipper;
    @Mock
    private
    GameContract.View view;

    @Mock
    GameRepository gameRepository;
    private GameContract.Presenter gamePresenter;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        gamePresenter = new GamePresenter(gameRepository, view);
    }

    @Test
    public void t1() {
        gamePresenter.start();

        verify(view).showLoadingView();
    }

    @Test
    public void onIemClicked() {
        gamePresenter.onItemClicked(0, card, viewFlipper);

    }

    @Test
    public void removeCardsFromMaps() {
        gamePresenter.removeCardsFromMaps();
        verify(view).removeViewFlipper();
    }

}

File: app/src/test/java/kittentrate/ui/game/GamePresenterTest.kt
package kittentrate.ui.game

/**
 * Created by Manuel Lorenzo
 */
//@RunWith(RobolectricTestRunner::class)
//class GamePresenterTest : RobolectricTestHelper() {
//    @Mock
//    private lateinit var view: GameContract.View
//    @Mock
//    private lateinit var repository: Repository
//
//    private lateinit var gamePresenter: GameContract.Presenter
//
//    @Before
//    @Throws(Exception::class)
//    fun setUp() {
//        MockitoAnnotations.initMocks(this)
//        gamePresenter = GamePresenter(repository, view)
//    }
//
//    @Test
//    fun `view should start as None and after starting it should have a Loading state`() {
//        //verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.None>()
//        whenever(repository.photos).thenReturn(Observable.just(listOf()))
//        gamePresenter.start()
//        verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.Loading>()
//        verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.Success<List<PhotoEntity>>>()
//    }
//
//    @Test
//    fun `view should start as None and after starting it should have an Error state`() {
//        //verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.None>()
//        whenever(repository.photos).thenReturn(Observable.error(Throwable()))
//        gamePresenter.start()
//        verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.Loading>()
//        verify<GameContract.View>(view).networkViewState = isA<NetworkViewState.Error>()
//    }
//
//    @Test
//    fun `should remove view flipper when removing cards from maps`() {
//        gamePresenter.removeCardsFromMaps()
//        verify<GameContract.View>(view).removeViewFlipper()
//    }
//
//    @Test
//    fun `should check for game end after score changes`() {
//        gamePresenter.onGameScoreChanged(1)
//
//        verify<GameContract.View>(view).onScoreChanged(eq(1))
//        verify<GameContract.View>(view).checkGameFinished()
//    }
//}